aws_xray_sdk/core/recorder.py

```diff
@@ -296,15 +296,46 @@ def begin_subsegment(self, name, namespace='local'):
             log.warning("No segment found, cannot begin subsegment %s." % name)
             return None
 
-        if not segment.sampled:
+        current_entity = self.get_trace_entity()
+
+        if not current_entity.sampled:
             subsegment = DummySubsegment(segment, name)
+            subsegment.sampled = False
```

Comment:
This code is redundant. A `DummySubsegment` is already unsampled.
https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/dummy_entities.py#L96